### PR TITLE
fix relationship generation

### PR DESCRIPTION
**Pull request by ca-link**

### CODE REVIEW

File: Blueprint41/DatastoreTemplates/Domain_Data_Entity_Relation.tt [hunk 1]
```diff
@@ -194,7 +200,7 @@ namespace <#= Settings.FullCRUDNamespace #>
     }
 #>
 
-        public static Relationship Relationship => Threadsafe.LazyInit(ref _relationship, () => <#= Datastore.GetType().FullName.Replace("+", ".") #>.Model.Relations["<#= DALRelation.Name #>"]);
+        public static Relationship Relationship => ThreadSafe.LazyInit(ref _relationship, () => <#= Datastore.GetType().FullName.Replace("+", ".") #>.Model.Relations["<#= DALRelation.Name #>"]);
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/Domain_Data_Entity_Relation.tt:197]: changed to capital **"S"**

File: Blueprint41/DatastoreTemplates/Domain_Data_Entity_Relation.tt [hunk 2]
```diff
@@ -124,6 +124,12 @@ namespace <#= Settings.FullCRUDNamespace #>
         {
 #>
                 if (<#= attr.Name #>.HasValue) conditions.Add(alias.<#= attr.Name #> == <#= attr.Name #>.Value?.ToString());
+<#
+        }
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/Domain_Data_Entity_Relation.tt]: add checking if property type is an array

File: Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt [hunk 3]
```diff
@@ -999,6 +1011,12 @@ if (!DALModel.IsAbstract)
         {
 #>
                 if (<#= argument.Name #>.HasValue) conditions.Add(alias.<#= argument.Name #> == <#= argument.Name #>.Value?.ToString());
+<#
+        }
+        else if (argument.SystemReturnType.IsArray)
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt]: add checking if property type is an array

File: Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt [hunk 4]
```diff
@@ -890,6 +896,12 @@ if (!DALModel.IsAbstract)
...
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt]: add checking if property type is an array

File: Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt [hunk 5]
```diff
@@ -785,6 +785,12 @@ if (!DALModel.IsAbstract)
...
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/Domain_Data_Entity.tt]: add checking if property type is an array

File: Blueprint41/DatastoreTemplates/GeneratorBase.cs [hunk 6]
```diff
@@ -165,6 +165,8 @@ protected string GetResultType(Type type)
                         return "StringListResult";
                     else
                         return "ListResult";
+                case "String[]":
```
Comment (ca-link) [inline Blueprint41/DatastoreTemplates/GeneratorBase.cs:168]: add checking if property type is an array of string

**Review by circles-arrows — APPROVED** [review]

**Pull request merged**